hs_core/tasks.py

```diff
@@ -398,6 +402,26 @@ def create_bag_by_irods(resource_id):
 
     res = get_resource_by_shortkey(resource_id)
     istorage = res.get_irods_storage()
+    bag_full_name = res.get_irods_path('bags/{res_id}.zip'.format(res_id=resource_id), prepend_short_id=False)
+
+    if res.locked:
+        # resource is locked meaning its bag is being created by another celery task, wait until the lock
+        # is released which indicates the bag is available
+        while res.locked:
+            sleep(1)
+
+        # lock is released, check if resource bag indeed exists
+        if istorage.exists(bag_full_name):
+            # the bag is made available by another celery task
+            return True
+        else:
+            # the lock is released, but resource bag does not exist, which indicates bag generation failure
+            return False
+    else:
+        with transaction.atomic():

```

Comment:
You're only setting one field.  transaction.atomic won't do anything here.

Reply:
@sblack-usu That is a direction I am trying to go. I'll let you know if I need help with it. Thanks